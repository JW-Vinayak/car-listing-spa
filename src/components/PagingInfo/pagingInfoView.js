import React from "react";
import { connect } from "react-redux";

const PagingInfo = props => {
  const { currentPage, totalPages, totalRecords } = props;
  const recordsPerPage = 10;
  console.log(
    currentPage,
    recordsPerPage,
    totalPages,
    currentPage * recordsPerPage
  );
  const fromPage = (currentPage - 1) * recordsPerPage + 1;
  let toPage = 0;

  console.log("paging props", currentPage === totalPages, fromPage);

  if (currentPage === totalPages) {
    toPage = fromPage + (totalRecords % 10) - 1;
  } else {
    toPage = fromPage + 9;
  }

  return (
    <div className="paging-info">
      <div className="text bold">Available Cars</div>
      <div className="text">
        Showing {fromPage}-{toPage} of {totalRecords} results
      </div>
    </div>
  );
};

const mapStateToProps = state => ({
  currentPage: state.pagination.currentPage,
  totalPages: state.cars.totalPages,
  totalRecords: state.cars.totalRecords
});

export default connect(mapStateToProps)(PagingInfo);
