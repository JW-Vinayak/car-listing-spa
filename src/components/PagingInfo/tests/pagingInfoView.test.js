import React from "react";
import thunk from "redux-thunk";
import PagingInfo from "../pagingInfoView";
import configureMockStore from "redux-mock-store";
import renderer from "react-test-renderer";

const middlewares = [thunk];

const mockStore = configureMockStore(middlewares);

let store = mockStore({
  pagination: {
    currentPage: 1
  },
  cars: {
    totalPages: 8,
    totalRecords: 76
  }
});

describe("PagingInfo component", () => {
  it("shallow renders without crashing", () => {
    const tree = renderer.create(<PagingInfo store={store} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
