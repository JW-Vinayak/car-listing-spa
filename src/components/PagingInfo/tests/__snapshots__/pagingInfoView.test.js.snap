// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PagingInfo component shallow renders without crashing 1`] = `
<div
  className="paging-info"
>
  <div
    className="text bold"
  >
    Available Cars
  </div>
  <div
    className="text"
  >
    Showing 
    1
    -
    10
     of 
    76
     results
  </div>
</div>
`;
